Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -266,9 +273,41 @@ extension Compiler.ByteCodeGen {
     }
   }
 
+  mutating func resolveReferences() throws {
+    for (id, uses) in unresolvedReferences {
+      guard let offset = captureOffsets[id] else {
+        throw RegexCompilationError.uncapturedReference
+      }
+      for use in uses {
+        builder.instructions[use.rawValue] =
+          Instruction(.backreference, .init(capture: .init(offset)))
+      }
+    }
+  }
+
+  mutating func emitUnresolvedReference(id: Reference.ID) {
```

Comment:
Not sure if this is future-proof, as this approach assumes we never insert to the middle of the instruction list.